src/ui.rs

```
@@ -31,111 +36,227 @@ fn handle_keypress(
+        // Not sure why you'd want to use the Top layer instead, but here's the syntax.
+        // gtk_layer_shell::LayerShell::set_layer(&window, gtk_layer_shell::Layer::Top);
+        gtk_layer_shell::LayerShell::set_keyboard_mode(&window, gtk_layer_shell::KeyboardMode::Exclusive);
+        gtk_layer_shell::LayerShell::set_anchor(&window, gtk_layer_shell::Edge::Top, true);
+        gtk_layer_shell::LayerShell::set_anchor(&window, gtk_layer_shell::Edge::Bottom, true);
+        gtk_layer_shell::LayerShell::set_anchor(&window, gtk_layer_shell::Edge::Left, true);
+        gtk_layer_shell::LayerShell::set_anchor(&window, gtk_layer_shell::Edge::Right, true);
+
+        // Set monitor for this output
+        let display = gtk4::gdk::Display::default().unwrap();
+        let monitors = display.monitors();
+        for i in 0..monitors.n_items() {
+            if let Some(monitor) = monitors.item(i).and_then(|obj| obj.downcast::<gtk4::gdk::Monitor>().ok()) {
+                let geometry = monitor.geometry();
+                if geometry.x() <= output.rect.x && 
+                   output.rect.x < geometry.x() + geometry.width() &&
+                   geometry.y() <= output.rect.y &&
+                   output.rect.y < geometry.y() + geometry.height() {
+                    gtk_layer_shell::LayerShell::set_monitor(&window, Some(&monitor));
+                    break;
+                }
+            }
+        }
+
+        let fixed = gtk4::Fixed::new();
+        let mut local_key_to_con_id = HashMap::new();
+
+        // Create labels for windows
+        for (_idx, window_node) in windows.iter().enumerate() {
+            let (x, y) = calculate_geometry(window_node, &output, args.clone());
+            let label = gtk4::Label::new(Some(""));
+
+            let letter = match chars.next() {
+                Some(c) => c,
+                None => 'a', // Fallback if we run out of characters
```

I think I prefer panicking if we run out of characters rather than falling back to 'a,' since the behavior isn't
entirely obvious. In particular, i'm not sure if it's obvious which window is the real 'a', i.e. the one that gets focused.

---

I'll push an update along these lines.